lib/auth/auth_with_roles.go

```diff
@@ -2846,15 +2849,20 @@ func (a *ServerWithRoles) DeleteOIDCConnector(ctx context.Context, connectorID s
 
 // UpsertSAMLConnector creates or updates a SAML connector.
 func (a *ServerWithRoles) UpsertSAMLConnector(ctx context.Context, connector types.SAMLConnector) error {
+	if !modules.GetModules().Features().SAML {
+		// TODO(zmb3): ideally we would wrap ErrREquiresEnterprise here, but
+		// we can't currently propagate wrapped errors across the gRPC boundary,
+		// and we want tctl to display a clean user-facing messasge in this case
+		return trace.Wrap(ErrSAMLRequiresEnterprise)
```

Comment:
The comment above says that "ideally we'd wrap ErrRequiresEnterprise" but this is exactly what we're doing here, no?